spec.emu

```diff
@@ -262,6 +337,58 @@ contributors: Mark S. Miller, Richard Gibson
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-%typedarray%.prototype.filter" number="10">
+        <h1>%TypedArray%.prototype.filter ( _callback_ [ , _thisArg_ ] )</h1>
+        <p>The interpretation and use of the arguments of this method are the same as for `Array.prototype.filter` as defined in <emu-xref href="#sec-array.prototype.filter"></emu-xref>.</p>
+        <p>This method performs the following steps when called:</p>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. Let _taRecord_ be ? ValidateTypedArray(_O_, ~seq-cst~).
+          1. Let _len_ be TypedArrayLength(_taRecord_).
+          1. If IsCallable(_callback_) is *false*, throw a *TypeError* exception.
+          1. Let _kept_ be a new empty List.
+          1. Let _captured_ be 0.
+          1. Let _k_ be 0.
+          1. Repeat, while _k_ &lt; _len_,
+            1. Let _Pk_ be ! ToString(𝔽(_k_)).
+            1. Let _kValue_ be ! Get(_O_, _Pk_).
+            1. Let _selected_ be ToBoolean(? Call(_callback_, _thisArg_, « _kValue_, 𝔽(_k_), _O_ »)).
+            1. If _selected_ is *true*, then
+              1. Append _kValue_ to _kept_.
+              1. Set _captured_ to _captured_ + 1.
+            1. Set _k_ to _k_ + 1.
+          1. Let _A_ be ? TypedArraySpeciesCreate(_O_, « 𝔽(_captured_) »<ins>, ~write~</ins>).
+          1. Let _n_ be 0.
+          1. For each element _e_ of _kept_, do
+            1. Perform <del>!</del><ins>?</ins> Set(_A_, ! ToString(𝔽(_n_)), _e_, *true*).
```

Comment:
Isn't this still infallible? The other changes from this PR ensure that `A` is large enough, has a non-detached buffer, and is not immutable.

Reply:
Yep, this was obviated by 9d03c0d4d2aca4328ad6fd351d358ce4bc3f41b1. Fixed.